staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go

```diff
@@ -191,7 +192,18 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 	s.GenericAPIServer.Handler.NonGoRestfulMux.Handle("/apis", crdHandler)
 	s.GenericAPIServer.Handler.NonGoRestfulMux.HandlePrefix("/apis/", crdHandler)
 
-	crdController := NewDiscoveryController(s.Informers.Apiextensions().InternalVersion().CustomResourceDefinitions(), versionDiscoveryHandler, groupDiscoveryHandler)
+	// This will wire up openapi for extension api server
+	s.GenericAPIServer.PrepareRun()
```

Comment:
this does not belong here. It is to be called shortly before run as a last chance before run to wire things.

Reply:
Good point. I put it here because I need the openapi route installation ready, before passing the openapi services and spec to a controller that watches CRD and operates on the openapi services w/ spec.

Maybe we can do either of:
1. separate [openapi route installation](https://github.com/kubernetes/kubernetes/blob/f48e18faa4dc035cc927c6a2b34c83c8475b55fa/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go#L237-L241) out from `PrepareRun`, so we can wire up openapi earlier but keep everything else as it used to be: I'm not sure if it's safe/logically-correct to do so
2. use/create another controller that watches CRD after things are [wired up](https://github.com/kubernetes/kubernetes/blob/f48e18faa4dc035cc927c6a2b34c83c8475b55fa/cmd/kube-apiserver/app/server.go#L180-L185) for extension api server: I think it's similar to one of my previous approach. It has the problem of having more CRD-related code out of k8s.io/apiextension-apiserver package.